### PR TITLE
Currency converted

### PR DESCRIPTION
**Pull request by Carlos-Diaz-07**
Now is ready to use you just need to require Convert in the controller.
require

To use:
use
Value is the portfolio value in this case but it can be anything as long as it's a float, api_currency and user currency are passed next and it should always be the currency symbol in capital letters ex. "USD"

if we need to change the refreshing rate just go to the currency_canverter.rb file and change it here
change refreshing time

Lastly the rates get save in a XML file that is kept next to the currency_converter.rb
location

### CODE REVIEW

**Comment by andrerferrer** [comment]
Great PR description

File: lib/currency_converter.rb [hunk 1]
```diff
@@ -0,0 +1,26 @@
+require 'eu_central_bank'
+class Convert
+  def currency(quantity, currency_in, currency_out)
```
Comment (andrerferrer) [inline lib/currency_converter.rb:3]: I would make it a class method

File: lib/currency_converter.rb [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+    # reads the rates from the specified location
+    bank.update_rates(cache)
+
+    # Here we update the rates once a day or you can change it if needed
+    if !bank.rates_updated_at || bank.rates_updated_at < Time.now - 1.days
+      bank.save_rates(cache)
+      bank.update_rates(cache)
+    end
+
+    # Now we do the exchange from value of currency#A to currency#B
```
Comment (andrerferrer) [inline lib/currency_converter.rb:23]: RLY good comments

**Review by andrerferrer — COMMENTED** [review]
Great work.
Rly rly well implemented, well written and documented.

I would have made it be a Service Object instead of a class in the lib folder, though.

[Service Objects](https://github.com/andrerferrer/scraping-demo) - How to implement a service object.

File: app/controllers/portfolios_controller.rb [hunk 3]
```diff
@@ -1,4 +1,5 @@
 class PortfoliosController < ApplicationController
+  require 'currency_converter'
```
[the remark this answers is not in the document]
Comment (Carlos-Diaz-07) [reply app/controllers/portfolios_controller.rb:2]: Yes, It seems to be necessary. I tried without it and it breaks.

File: app/controllers/portfolios_controller.rb [hunk 4]
```diff
@@ -23,7 +24,9 @@ def create
 
   def show
     authorize @portfolio
-    @portfolio_value = investment_price(portfolio_symbols)
+    value = investment_price(portfolio_symbols)
+    api_curr = "USD"
```
[the remark this answers is not in the document]
Comment (Carlos-Diaz-07) [reply app/controllers/portfolios_controller.rb:28]: I just added a comment

File: app/controllers/portfolios_controller.rb (hunk 3 above)
[the remark this answers is not in the document]
Comment (andrerferrer) [reply app/controllers/portfolios_controller.rb:2]: It's because of the lib folder

File: lib/currency_converter.rb (hunk 1 above)
Comment (Carlos-Diaz-07) [reply lib/currency_converter.rb:3]: Now changed to instance class

File: lib/currency_converter.rb (hunk 2 above)
Comment (Carlos-Diaz-07) [reply lib/currency_converter.rb:23]: Thank you!!

File: app/controllers/portfolios_controller.rb (hunk 3 above)
[the remark this answers is not in the document]
Comment (Carlos-Diaz-07) [reply app/controllers/portfolios_controller.rb:2]: Should I put it somewhere else?

[the remark this answers is not in the document]
Comment (andrerferrer) [reply app/controllers/portfolios_controller.rb:2]: @Carlos-Diaz-07 check this https://github.com/coorsa/assetbase/pull/50#pullrequestreview-900959928

[the remark this answers is not in the document]
Comment (Carlos-Diaz-07) [reply app/controllers/portfolios_controller.rb:2]: I see thank you. I'll have a look

**Pull request merged**